sympy/polys/matrices/domainmatrix.py

```diff
@@ -52,7 +52,10 @@ def from_Matrix(cls, M, **kwargs):
         return cls.from_list_sympy(*M.shape, M.tolist(), **kwargs)
 
     @classmethod
-    def get_domain(cls, items_sympy, **kwargs):
+    def get_domain(cls, items_sympy, radicals_limit=None, **kwargs):
+        if radicals_limit == 1:
+            kwargs['extension'] = True
```

Comment:
The idea of the `radicals_limit` option for this function would be that we count the number of radicals here and then compare that with `radicals_limit` which could be passed as `radicals_limit=1` when calling `get_domain`. You seem to have this backwards that the caller counts the radicals and this function compares that with 1.

Reply:
Sure, kindly review the latest commit.
Thanks.